src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/ExtendedFortisEvent.scala

```diff
@@ -0,0 +1,21 @@
+package com.microsoft.partnercatalyst.fortis.spark.analyzer
+
+import java.util.UUID
+
+import com.microsoft.partnercatalyst.fortis.spark.dto.{Analysis, Details, FortisEvent, Location}
+
+case class ExtendedFortisEvent[T](
+  details: ExtendedDetails[T],
+  analysis: Analysis
+) extends FortisEvent
+
+case class ExtendedDetails[T](
+  id: UUID,
+  createdAtEpoch: Long,
+  body: String,
+  title: String,
+  publisher: String,
+  sourceUrl: String,
+  sharedLocations: List[Location] = List(),
+  original: T
```

Comment:
As I'd commented on that discussion, there are a lot of issues with assuming we can converge every unique data model into a single unanalyzed representation.

I'd mentioned that we needed to *override* sentiment analysis for Tadaweb, and to support this, the old implementation was modified to *always* override if an analysis was already present. That's also not necessarily correct, since perhaps another pipeline will need to only augment the analysis.

The current `TextPipeline` is already getting hairy. Concrete `Analyzer`s that can handle analysis how they like allows for cleaner common code and more flexibility/extensibility. Fortis will one day be maintained by the community, and so we should design with extensibility in mind.

Serializing a few extra properties should not be an issue, since this will only happen once per event, given that our pipeline is largely sequential. See my comment above on line `97` of `Pipeline.scala`.